labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -37,15 +41,29 @@ object RecursionPatternMatchingExercise {
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
   def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case s if s.length == 1 => List(s)
+      case s::rest => if (s == rest.head){
+          var res = groupConsecutive(rest)
+          List(res.head ++ List(s)) ++ res.tail
+      }
+      else {
+        List(List(s)) ++ groupConsecutive(rest)
+      }
+    }
   }
+  
 
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
   def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case List() => List()
+      case s if s.length == 1 => List(s)
+      case s::rest => List(List(s) ++ rest.filter((a) => a == s)) ++ groupEquals(rest.filter((a) => a != s))
```

Comment:
I did it like this (in filter (_ == x)) :: groupEquals(in filterNot (_ == x)). I think it might be better to straight filter 'in' rather than rebuild it and filter on a smaller part.